sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamQueryStore.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.streaming
+
+import java.util.UUID
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql.streaming.StreamingQuery
+
+/**
+ * A class that holds [[StreamingQuery]] active across all sessions to manage the lifecycle
+ * of the stream.
+ */
+private[sql] class StreamQueryStore {
+  private val activeStreamingQueries = new ConcurrentHashMap[UUID, (StreamingQuery, Long)]()
+  // There maybe more than one inactive stream query with same query ID, as we can run same
+  // stream query many times after it failed or terminated.
+  private val inactiveStreamingQueries = new ConcurrentHashMap[(UUID, Long), StreamingQuery]()
```

Comment:
I'd like to hear the voices of tracking inactive streaming queries (with details), as it brings the code complexity as well as query information will be no longer be cleaned up. Technically, the only reason of introducing StreamQueryStore is tracking inactive streaming queries - otherwise existing `activeStreamingQueries` can serve the necessary features well.

Reply:
Yeah, tracking inactive streaming queries is used to provide the error details. It will be helpful when we submit multiple stream query into `Spark Thrifter Server`. It helps end-users to see why query filed.

Reply:
IMO, tracking inactive streaming queries is necessary, but we still need to have some strategies for cleanup, e.g., add a config for the max inactive streaming queries number.
Also, is there any other way to keep a streaming query status rather than store the whole StreamingQuery instance here? The instance contains even the execution plan, which would be memory consuming.